src/Eventso.Subscription.Kafka/DeadLetter/TopicRetryingService.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using Eventso.Subscription.Kafka.DeadLetter.Store;
+using Microsoft.Extensions.Logging;
+
+namespace Eventso.Subscription.Kafka.DeadLetter
+{
+    public sealed class TopicRetryingService
+    {
+        private readonly string _topic;
+        private readonly IPoisonEventStore _poisonEventStore;
+        private readonly IDeserializer<ConsumedMessage> _deserializer;
+        private readonly IEventHandler<Event> _eventHandler;
+        private readonly ILogger<TopicRetryingService> _logger;
+
+        public TopicRetryingService(
+            string topic,
+            IPoisonEventStore poisonEventStore,
+            IDeserializer<ConsumedMessage> deserializer,
+            IEventHandler<Event> eventHandler,
+            ILogger<TopicRetryingService> logger)
+        {
+            _topic = topic;
+            _poisonEventStore = poisonEventStore;
+            _deserializer = deserializer;
+            _eventHandler = eventHandler;
+            _logger = logger;
+        }
+
+        public async Task Retry(CancellationToken cancellationToken)
+        {
+            using var retryScope = _logger.BeginScope(
+                new[] { new KeyValuePair<string, string>("eventso_retry_topic", _topic) });
+
+            _logger.LogInformation($"Started event retrying.");
+
+            using var events = new PooledList<Event>(4);
+
+            await foreach (var storedEvent in _poisonEventStore.GetEventsForRetrying(_topic, cancellationToken))
+            {
+                var @event = Deserialize(storedEvent);
+                events.Add(@event);
+
+                _logger.LogInformation($"Queued event {@event.GetTopicPartitionOffset()} for retrying.");
+            }
+
+            await _eventHandler.Handle(events, cancellationToken);
+
+            _logger.LogInformation($"Finished event retrying.");
+        }
+
+        private Event Deserialize(StoredPoisonEvent storedEvent)
+        {
+            var headers = new Headers();
+            foreach (var header in storedEvent.Headers)
+                headers.Add(header.Key, header.Data.ToArray());
+
+            var consumeResult = new ConsumeResult<Guid, ConsumedMessage>
+            {
+                // shaky and depends on Confluent.Kafka contract
```

Comment:
so dlq will only work for kafka? PoisonEventRetryingHost depends on this service, which depends on kafka

Reply:
i understand it is a little bit confusing but the whole Hosting assembly is for kafka. i'm thinking about moving it inside Kafka assembly to fix it